packages/paste-core/components/sidebar/src/SidebarHeader.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from '@twilio-paste/box';
+import {Button} from '@twilio-paste/button';
+import type {BoxProps} from '@twilio-paste/box';
+import {ProductHomeIcon} from '@twilio-paste/icons/esm/ProductHomeIcon';
+
+import {SidebarContext} from './SidebarContext';
+
+export interface SidebarHeaderProps extends React.HTMLAttributes<HTMLButtonElement> {
+  productIcon?: React.ReactNode;
+  onClick?: () => void;
+  element?: BoxProps['element'];
+}
+
+export const SidebarHeader = React.forwardRef<HTMLButtonElement, SidebarHeaderProps>(
+  ({element = 'SIDEBAR_HEADER', productIcon, onClick, children}, ref) => {
+    const {collapsed} = React.useContext(SidebarContext);
+
+    return (
+      <Box
+        padding="space60"
+        borderBottomWidth="borderWidth10"
+        borderBottomStyle="solid"
+        borderBottomColor="colorBorderInverseWeaker"
+        display="flex"
+        alignItems="center"
+        justifyContent="flex-start"
+        columnGap="space50"
+        overflow="hidden"
+        element={element}
+        ref={ref}
+      >
+        <Button size="icon" variant="inverse" element={`${element}_BUTTON`} onClick={onClick}>
+          {productIcon ? productIcon : <ProductHomeIcon size="sizeIcon20" decorative element={`${element}_ICON`} />}
+        </Button>
+        <Box
+          opacity={collapsed ? '0' : '1'}
+          element={`${element}_TEXT`}
+          fontSize="fontSize30"
+          fontWeight="fontWeightSemibold"
+          lineHeight="lineHeight30"
+          letterSpacing="-.02em"
+          as="h5"
```

Comment:
question: why did you go with h5

Reply:
yeah, I might have some caution on the use of ay heading level given document hierarchy will be unknown. We might just get away with a div here